### DISCUSSION TITLE
用pycharm调试

### DISCUSSION THREAD

**Discussion by YIYEYIYE** [FederatedAI/FATE#5658] [Q&A]
1.安装了pycharm，怎么将fate作为项目添加进去并调试？
2.pycharm是不是只能安专业版的？
3.有没有用pycharm调试FATE的教程？

**Comment by PenglinYang** [comment]
配置一下pycharm的environment就行了。environment参数可以先命令行按照官方文档run一遍，把env的变量拷贝到pycharm 中。debug就行了

**Comment by Ariel08081214** [comment]
> 配置一下pycharm的environment就行了。environment参数可以先命令行按照官方文档run一遍，把env的变量拷贝到pycharm 中。debug就行了

请问是远程连接Linux还是直接在Windows呢？@PenglinYang

**Comment by YIYEYIYE** [comment]
已经会了，用的虚拟机。感谢！




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年7月25日(星期二) 中午11:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FederatedAI/FATE] 用pycharm调试 (Issue #4367)





    
1.安装了pycharm，怎么将fate作为项目添加进去并调试？ 2.pycharm是不是只能安专业版的？ 3.有没有用pycharm调试FATE的教程？
   
配置一下pycharm的environment就行了。environment参数可以先命令行按照官方文档run一遍，把env的变量拷贝到pycharm 中。debug就行了
  
请问是远程连接Linux还是直接在Windows呢？

**Comment by Ariel08081214** [comment]
> 



> 已经会了，用的虚拟机。感谢！

请教个问题，你下载的是docker包还是编译后的源码来debug呢？@YIYEYIYE

**Comment by YIYEYIYE** [comment]
我记得是docker




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年7月27日(星期四) 上午9:43
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FederatedAI/FATE] 用pycharm调试 (Issue #4367)





    
已经会了，用的虚拟机。感谢！
  
***@***.***

**Comment by PenglinYang** [comment]
# fate1.9.1 在pycharm中调试：
## 1,首先使用github中的部署和运行命令把fateserver run起来：
  cd ${FATE_PROJECT_BASE};
  source bin/init_env.sh;
  cd fateflow;
  bash bin/service.sh status;
  bash bin/service.sh start
## 2,通过上述命令得到init_env.sh,里边有该项目所需的环境变量；通过在T中输入env，得到fate项目设置的环境变量
## 3，pycharm edit configuration，python下新建条目，将环境变量输入到environment中：
  EGGROLL_LOG_LEVEL=INFO]
  FATE_DEPLOY_BASE=/home/suede/1FATE/FATE
  FATE_LOG_LEVEL=DEBUG
  FATE_PROFILE_LOG_ENABLE=0
  FATE_PROJECT_BASE=/home/suede/1FATE/FATE
  FATE_VENV_BASE=/home/suede/fate_venv/fatevenv/
  PYTHONPATH=/home/suede/1FATE/FATE/python:/home/suede/1FATE/FATE/fateflow/python
  PYTHONUNBUFFERED=1

启动文件 __main__ /FATE/fateflow/python/fate_flow/fate_flow_server.py 
# fateflow基于flask实现，
fateflow server 通过start启动多线程
run_simple(apps)启动flask web server服务

@Ariel881214

**Comment by Ariel08081214** [comment]
> # fate1.9.1 在pycharm中调试：
> ## 1,首先使用github中的部署和运行命令把fateserver run起来：
> cd ${FATE_PROJECT_BASE}; source bin/init_env.sh; cd fateflow; bash bin/service.sh status; bash bin/service.sh start
> 
> ## 2,通过上述命令得到init_env.sh,里边有该项目所需的环境变量；通过在T中输入env，得到fate项目设置的环境变量

非常感谢 。请问第一步是参考《源码部署FATE单机版》这个文档吗？

**Comment by PenglinYang** [comment]
是的
